lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -670,6 +689,25 @@ if Enum.all?(
       }
     end
 
+    defp filter_placements({Pad.ref(_type, id), _placement}, track),
+      do: id == track.id or id == {:blank, track.id}
+
+    defp update_layout_action(_track, _video_layout, %{mixer_config: nil} = state),
+      do: {[], state}
+
+    defp update_layout_action(%{type: :audio}, _video_layout, state), do: {[], state}
+
+    defp update_layout_action(track, :removed, state) do
+      {placements, video_layout} =
+        state.mixer_config.video.layout_module.track_removed(state.video_layout, track)
+
+      state = %{state | video_layout: video_layout}
+      update_layout_action(track, placements, state)
+    end
```

Comment:
I feel like this clause should be inlined. Passing `:removed` when in all other cases there is `video_layout` and calling itself recursively makes it hard to understand